portability-transfer/src/main/java/org/datatransferproject/transfer/InMemoryIdempotentImportExecutor.java

```diff
@@ -17,31 +17,41 @@
 package org.datatransferproject.transfer;
 
 import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
 import org.datatransferproject.api.launcher.Monitor;
 import org.datatransferproject.spi.transfer.provider.IdempotentImportExecutor;
+import org.datatransferproject.types.transfer.errors.ErrorDetail;
 
 import java.io.IOException;
 import java.io.Serializable;
+import java.util.Collection;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.concurrent.Callable;
 
-/**
- * A {@link IdempotentImportExecutor} that stores known values in memory.
- */
+/** A {@link IdempotentImportExecutor} that stores known values in memory. */
 public class InMemoryIdempotentImportExecutor implements IdempotentImportExecutor {
-  final private Map<String, Serializable> knownValues = new HashMap<>();
+  private final Map<String, Serializable> knownValues = new HashMap<>();
+  private final Map<String, ErrorDetail> errors = new HashMap<>();
   private final Monitor monitor;
 
   public InMemoryIdempotentImportExecutor(Monitor monitor) {
     this.monitor = monitor;
   }
 
   @Override
-  public <T extends Serializable> T execute(
-      String idempotentId,
-      String itemName,
-      Callable<T> callable) throws IOException {
+  public <T extends Serializable> T executeAndSwallowExceptions(
+      String idempotentId, String itemName, Callable<T> callable) throws IOException {
+    try {
+      return executeOrThrowException(idempotentId, itemName, callable);
+    } catch (Exception e) {
+      return null;
```

Comment:
So the executeOrThrowException throw IOException.  Do we want to catch that and ignore but also have a block for a general Exception in case anything else bubbles up, especially if there are future additions to this code.

Reply:
Ya I changed it so any exceptions that are throw in the throwing catch block get bubbled up.  Hopefully that never happens but it will be much less confusing if it does happen now.